src/Validation/AbstractValidators.php

```diff
@@ -231,7 +233,7 @@ public function update($record, array $document): ValidatorInterface
     {
         return $this->validatorForResource(
             $this->dataForUpdate($record, $document),
-            $this->rules($record),
+            $this->rules($record, $document),
```

Comment:
Same change here: assign `$data = $this->dataForUpdate($record, $document)`, then pass `$data` to the `rules()` method.